src/ixdat/readers/ec_ms.py

```diff
@@ -1,9 +1,73 @@
 from . import TECHNIQUE_CLASSES
+import pickle
+from ..data_series import TimeSeries, ValueSeries
+from ..measurements import Measurement
+from ..techniques.ec_ms import ECMSMeasurement
+
 
 ECMSMeasruement = TECHNIQUE_CLASSES["EC-MS"]
 
 
 class EC_MS_CONVERTER:
-    def read(self):
-        # return ECMSMeasurement(**obj_as_dict)
-        pass
+"""Class that converts old .pkl files obtained from the legacy EC-MS package
+into a ixdat ECMSMeasurement object. Metadata from the .pkl is omitted.
+"""
+    def __init__(self):
+        print('Reader of old ECMS .pkl files')
+
+    def read(self, file_path):
+        """Return an ECMSMeasurement with the data recorded in path_to_file
+
+        This loops through the keys of the EC-MS dict and searches for MS and
+        EC data. Names the dataseries according to their names in the original
+        dict. Omitts any other data as well as metadata.
+
+        Args:
+            path_to_file (Path): The full abs or rel path including the
+            ".pkl" extension.
+        """
+        with open(file_path, "rb") as f:
+            data = pickle.load(f)
+
+        cols_str = data['data_cols']
+        cols_list = []
+
+        for col in cols_str:
+            if col[0] == "M" and col[-1] == "x":
```

Comment:
Drop the test for starting with "M" and use `endswith("-x")`, for reasons described below in more detail below when you deal with "-y"